### DISCUSSION TITLE
audacity  /BUILDING.md for Linux & Other OS

### DISCUSSION THREAD

**Discussion by mahnac72** [audacity/audacity#4694] [General]
I have never attempted this process before

From https://github.com/audacity/audacity/blob/master/BUILDING.md
Section for Linux & Other OS

I get as far as step 5. Installing Audacity

$ cd <build directory> fro which I /build/Debug/bin/Test34$ 
Then
$ sudo make install
fails with 
make: *** No rule to make target 'install'.  Stop
and I have no idea or can find what this means to resolve.

I tried to use the debug version of Audacity which runs in the background but when running install... issues.

colin@colin-MS-7B84:~/Downloads$ sudo ./audacity
[sudo] password for colin: 

(process:33019): Gdk-CRITICAL **: 06:44:57.072: IA__gdk_screen_get_root_window: assertion 'GDK_IS_SCREEN (screen)' failed

(process:33019): Gdk-CRITICAL **: 06:44:57.072: _gdk_pixmap_new: assertion '(drawable != NULL) || (depth != -1)' failed
/_work/conan-recipes/conan-recipes/.conan/data/wxwidgets/3.1.3.4-audacity/_/_/build/10f2526f12f71d8eb85afe3a18060ca4da9e690e/source_subfolder/src/gtk/bitmap.cpp(827): assert ""IsOk()"" failed in ConvertToImage(): invalid bitmap
/_work/conan-recipes/conan-recipes/.conan/data/wxwidgets/3.1.3.4-audacity/_/_/build/10f2526f12f71d8eb85afe3a18060ca4da9e690e/source_subfolder/src/common/image.cpp(2172): assert ""IsOk()"" failed in SetMaskColour(): invalid image
/_work/conan-recipes/conan-recipes/.conan/data/wxwidgets/3.1.3.4-audacity/_/_/build/10f2526f12f71d8eb85afe3a18060ca4da9e690e/source_subfolder/src/common/image.cpp(2223): assert ""IsOk()"" failed in SetMask(): invalid image
/_work/conan-recipes/conan-recipes/.conan/data/wxwidgets/3.1.3.4-audacity/_/_/build/10f2526f12f71d8eb85afe3a18060ca4da9e690e/source_subfolder/src/common/image.cpp(1853): assert ""IsOk()"" failed in GetWidth(): invalid image
/_work/conan-recipes/conan-recipes/.conan/data/wxwidgets/3.1.3.4-audacity/_/_/build/10f2526f12f71d8eb85afe3a18060ca4da9e690e/source_subfolder/src/common/image.cpp(1860): assert ""IsOk()"" failed in GetHeight(): invalid image
/_work/conan-recipes/conan-recipes/.conan/data/wxwidgets/3.1.3.4-audacity/_/_/build/10f2526f12f71d8eb85afe3a18060ca4da9e690e/source_subfolder/src/common/image.cpp(2112): assert ""IsOk()"" failed in GetAlpha(): invalid image
/_work/conan-recipes/conan-recipes/.conan/data/wxwidgets/3.1.3.4-audacity/_/_/build/10f2526f12f71d8eb85afe3a18060ca4da9e690e/source_subfolder/src/common/image.cpp(2232): assert ""IsOk()"" failed in HasMask(): invalid image
/_work/conan-recipes/conan-recipes/.conan/data/wxwidgets/3.1.3.4-audacity/_/_/build/10f2526f12f71d8eb85afe3a18060ca4da9e690e/source_subfolder/src/common/image.cpp(1986): assert ""IsOk()"" failed in GetData(): invalid image

(process:33019): GdkPixbuf-CRITICAL **: 06:44:57.072: gdk_pixbuf_new_from_data: assertion 'data != NULL' failed

(process:33019): Gdk-CRITICAL **: 06:44:57.072: IA__gdk_screen_get_root_window: assertion 'GDK_IS_SCREEN (screen)' failed

(process:33019): Gdk-CRITICAL **: 06:44:57.072: IA__gdk_drawable_get_display: assertion 'GDK_IS_DRAWABLE (drawable)' failed

(process:33019): Gdk-CRITICAL **: 06:44:57.072: IA__gdk_cursor_new_from_pixbuf: assertion 'GDK_IS_DISPLAY (display)' failed

(process:33019): GLib-GObject-CRITICAL **: 06:44:57.072: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
Gtk-Message: 06:44:57.084: Failed to load module "canberra-gtk-module"
ALSA lib pcm_dsnoop.c:601:(snd_pcm_dsnoop_open) unable to open slave
ALSA lib pcm_dmix.c:1032:(snd_pcm_dmix_open) unable to open slave
ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.rear
ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.center_lfe
ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.side
Cannot connect to server socket err = No such file or directory
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
Cannot connect to server socket err = No such file or directory
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
ALSA lib pcm_oss.c:397:(_snd_pcm_oss_open) Cannot open device /dev/dsp
ALSA lib pcm_oss.c:397:(_snd_pcm_oss_open) Cannot open device /dev/dsp
ALSA lib pulse.c:242:(pulse_connect) PulseAudio: Unable to connect: Connection refused

ALSA lib pulse.c:242:(pulse_connect) PulseAudio: Unable to connect: Connection refused

ALSA lib confmisc.c:160:(snd_config_get_card) Invalid field card
ALSA lib pcm_usb_stream.c:482:(_snd_pcm_usb_stream_open) Invalid card 'card'
ALSA lib confmisc.c:160:(snd_config_get_card) Invalid field card
ALSA lib pcm_usb_stream.c:482:(_snd_pcm_usb_stream_open) Invalid card 'card'
ALSA lib pcm_dsnoop.c:601:(snd_pcm_dsnoop_open) unable to open slave
ALSA lib pcm_dmix.c:1032:(snd_pcm_dmix_open) unable to open slave
ALSA lib pcm_dmix.c:1032:(snd_pcm_dmix_open) unable to open slave
Cannot connect to server socket err = No such file or directory
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
06:44:58: Debug: ScreenToClient cannot work when toplevel window is not shown

**Comment by crsib** [comment]
`make install` should be invoked from the build directory.

**Reply by mahnac72** [reply]
Sorry to bother you with my inexperience. The process I followed was 

![image](https://github.com/audacity/audacity/assets/3999139/cd893094-75ea-4179-9e61-807ded7d26f2)

Which created the debug folder and the audacity executable file

![image](https://github.com/audacity/audacity/assets/3999139/fc49ebbc-b532-43cb-87eb-f567c9ff85d3)

I then copy this executable file to where I would normally Downloads and then invoke the 

make install directory I have the executable, this will place the install files in the correct places on my system? 

Am I on the correct track?

**Reply by mahnac72** [reply]
I normally use the terminal and from there would go make install ?

![image](https://github.com/audacity/audacity/assets/3999139/cb250b5c-13c2-4eab-a195-077232a47009)

**Comment by mahnac72** [comment]
Oops.. I found this now, I'll follow the readme and presumably this INSTALL once I have set as executable..

![image](https://github.com/audacity/audacity/assets/3999139/de0d5d0d-aaf8-4422-9e88-ce01f6a21668)

![Uploading image.png…]()

**Comment by mahnac72** [comment]
Nope that did not work either..

![image](https://github.com/audacity/audacity/assets/3999139/e4747834-895a-4fad-8777-7647de57fc98)

**Reply by LWinterberg** [reply]
INSTALL is a txt file, not a script.
